### ISSUE TITLE
Vim as an explicit IDE preference

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#9323]
We currently have access to workspaces via SSH, however adding Vim would make it a first-class citizen as part of the Gitpod experience. Alongside this we should consider what we can do to ensure things like dotfiles are also setup, and that workspaces potentially already load with vim + other similar tools such as tmux. This would support experiments such as https://github.com/gitpod-io/gitpod/issues/6707 where vim configuration can be presented to the user as part of onboarding.

**Comment by gtsiolis** [comment]
@loujaybee I haven't researched in depth about this yet, but here are some early designs on what I had in mind forhttps://github.com/gitpod-io/gitpod/issues/7713#issuecomment-1140534710. Cc @josepjaume

| Editors :a: | Editors :b: | Workspace Start |
|-|-|-|
| ![Editors](https://user-images.githubusercontent.com/120486/171058037-0814909d-cd9c-412c-a620-75dce43468d7.png) | <img width="1440" alt="Editors" src="https://user-images.githubusercontent.com/120486/171058041-68f86aaf-9cd0-4b29-8d76-91145e38b6fc.png"> | <img width="1440" alt="StartDropdown" src="https://user-images.githubusercontent.com/120486/171058363-54b74474-2569-443f-b021-9df8076da5c5.png"> |

**Comment by josepjaume** [comment]
I love this @gtsiolis! I would personally go for the first option for consistency with the rest of the IDE options :). Can't wait to be able to use this!

**Comment by fcsonline** [comment]
The first option looks great! And when the workspace starts, the user would be able to choose an application to open the ssh links instead Vscode.

**Comment by loujaybee** [comment]
@fcsonline - what application(s) are you opening for your local terminal, currently? Do you open just one, or multiple applications (like your terminal) on desktop? Having this option certainly removes some friction for command-line users, but I'm also curious about what other desktop applications you're configuring outside of the IDE / editor itself.

**Comment by axonasif** [comment]
> Right now none, but this would let me to associate a script to open a Terminal and enter to the Gitpod automatically. It would awesome.

That indeed would be very awesome!
I'm just thinking, in order for such a thing to work, gitpod will need to use a trigger a custom URL scheme registered on the users local machine to pass the ball to the user for deciding what to do 🤔

**Comment by axonasif** [comment]
> @loujaybee: alongside this we should consider what we can do to ensure things like dotfiles are also setup.

I think this is very relevant for ensuring a seamless experience as with VSCODE: https://github.com/gitpod-io/gitpod/issues/7592
You definitely don't want to wait for your (neo)vim setup or anything else to be finalized before your workspace can start.

While https://github.com/gitpod-io/gitpod/issues/8716 might also make sense for this case, so users could just mount their pre-build `.config/X`, `.cache/X` and etc. dirs across workspace.

**Comment by fcsonline** [comment]
> > Right now none, but this would let me to associate a script to open a Terminal and enter to the Gitpod automatically. It would awesome.
> 
> That indeed would be very awesome! I'm just thinking, in order for such a thing to work, gitpod will need to use a trigger a custom URL scheme registered on the users local machine to pass the ball to the user for deciding what to do 🤔

Exactly. With a custom schema, we would be able to assign an application/script to do whatever. I would great!

**Comment by sigmaSd** [comment]
Not stale

**Comment by henit-chobisa** [comment]
# Gitpod-NVIM from Your Menubar ✌️
Building upon @axonasif's awesome dotsh config! 🧡

- https://github.com/gitpod-samples/Gitpod-Raycast-Extension/issues/52

https://user-images.githubusercontent.com/73993394/232240942-a1479740-df09-4927-973c-e8794b64095d.mov

**Comment by PabloScolpino** [comment]
Hi!

It would complement this feature to have the generated SSH command include the list of port forwards defined in the `.gitpod.yml` file. Not sure if this was considered, but since the default is going to be the SSH connection string dialog, and nothing else is going to be launched in the user's workstation, port translation would need to be addressed.

**Comment by axonasif** [comment]
Hi @PabloScolpino, I think you're looking for https://www.gitpod.io/blog/local-app
If you keep the local-companion-app program running in the background, it will take care of port forwarding for you.

Also, shameless plug: A wrapper around SSH for port forwarding: https://github.com/axonasif/gssh

**Comment by loujaybee** [comment]
Not entirely resolving this issue, but definitely related: 

https://www.gitpod.io/changelog/code-in-the-terminal-without-leaving-your-browser!

Drop any feedback here: https://github.com/gitpod-io/gitpod/issues/17724

**Comment by jwpjrdev** [comment]
@loujaybee came here to say that!
